pyspecdata/core.py

```diff
@@ -2596,14 +2599,28 @@ def box_muller(length, return_complex=True):
 #}}}
 
 #{{{nddata
-#{{{ format out to a certain decimal place
-def dp(number,decimalplaces,scientific=False):
+def dp(number,decimalplaces=2,scientific=False,max_front=3):
+    """format out to a certain decimal places, potentially in scientific notation
+
+    Parameters
+    ----------
+    decimalplaces: int (optional, default 3)
+        number of decimal places
+    scientific: boolean (optional, default False)
+        use scientific notation
+    max_front: int (optional, default 3)
+        at most this many places in front of the decimal before switching
+        automatically to scientific notation.
+    """
```

Comment:
yes, this is nice

Reply:
Glad you like it, but I do not think I added it